views/package_suppliers.dt

```diff
@@ -0,0 +1,37 @@
+extends layout
+
+block title
+	- auto title = "Package suppliers";
+
+block body
+	h2 Introduction
+
+	p Dub supports package suppliers of type Dub Registry, Maven and File System. By default Dub will search on Dub Registry <a href="https://code.dlang.org">https://code.dlang.org</a> and various fallbacks.
+
+	p Package suppliers can be specified either using argument <code>--registry</code> or using setting <code>registryUrls</code>. The type of the package supplier could be specified using a prefix <code>dub+</code>, <code>mvn+</code> or <code>file://</code>. By default type Dub Registry is assumed.
+
+	h3 Dub Registry
+
+	p Dub packages are registerd at a Dub Registry server like <a href="https://code.dlang.org">https://code.dlang.org</a>. It isn't needed to prefix the address of a Dub Registry.
+
+		ul
+			li https://code.dlang.org/
+			li dub+https://code.dlang.org/
+
+	h3 Maven repository
+
+	p Dub can retrieve dub package zip archives located within a Maven repository like Nexus or Artifactory. The registry address needs to contain the Maven repository and a group id. The dub packages needs to be stored with a common group id, the dub package name as artifact id and dub version as artifact version. The dub package zip file must have the name <code>[packageName]-[version].zip</code>.
+
+	p Example
+
+	p Dub package zip archive <code>calculator-1.0.0.zip</code> is stored on Maven repository <a href="http://localhost:8040/maven/libs-release">http://localhost:8040/maven/libs-release</a> using group id <code>dubpackages</code>. As registry address <code>mvn+http://localhost:8040/maven/libs-release/dubpackages</code> needs to be specified.
+
+	p <code>dub fetch calculator@1.0.0 --skip-registry=all --registry=mvn+http://localhost:8040/maven/libs-release/dubpackages</code>
+
+	h3 File system
+
+	p A folder containing dub packages as zip files could be specified as registry. The zip files must have the name <code>[packageName]-[version].zip</code>.
+
+	ul
+		li Linux <code>dub fetch calculator@1.0.0 --skip-registry=all --registry=file:///var/dubpackages</code>
+		li Windows <code>dub fetch calculator@1.0.0 --skip-registry=all --registry=file://c:/d/dubpackages</code>
```

Comment:
Technically this `file://` URL is malformed. The formal syntax of a `file://` URL is `file://[host]/[path]` where host can be omitted and defaults to localhost. I think we should rather document this with triple-slash, even though we probably won't add file access on remote hosts via an unspecified protocol any time soon.

See https://tools.ietf.org/html/rfc1738#section-3.10

```suggestion
		li Windows <code>dub fetch calculator@1.0.0 --skip-registry=all --registry=file:///c:/d/dubpackages</code>
```

Reply:
fixed